src/libcore/slice/mod.rs

```diff
@@ -611,17 +574,18 @@ impl<T> [T] {
     #[inline]
     pub fn iter_mut(&mut self) -> IterMut<T> {
         unsafe {
-            let p = if mem::size_of::<T>() == 0 {
-                1 as *mut _
+            let ptr = self.as_mut_ptr();
+            assume(!ptr.is_null());
+
+            let end = if mem::size_of::<T>() == 0 {
+                (ptr as usize).wrapping_add(self.len()) as *mut _
```

Comment:
[Oh-so-long-ago](https://github.com/rust-lang/rust/pull/25434) the casting here between integers and pointers was found to inhibit optimizations when it comes to ZSTs, so would it be possible to use a similar strategy as that PR to avoid the pointer<->int conversions here?

Reply:
I just tried that, and it makes the tests fail...?!?